compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2394,6 +2448,25 @@ fn add_upstream_rust_crates<'a, B: ArchiveBuilder<'a>>(
                                 cmd.link_staticlib(name, verbatim);
                             }
                         }
+                        if sess.opts.unstable_opts.separate_native_rlib_dependencies {
+                            if let NativeLibKind::Static {
+                                bundle: Some(true) | None,
+                                whole_archive,
+                            } = lib.kind
+                            {
+                                let verbatim = lib.verbatim.unwrap_or(false);
+                                if whole_archive == Some(true) {
+                                    cmd.link_whole_staticlib(
+                                        name,
+                                        verbatim,
+                                        &[PathBuf::from(tmpdir)],
+                                    );
+                                } else {
+                                    cmd.link_staticlib(name, verbatim);
```

Comment:
```suggestion
                                    cmd.link_rlib(name, verbatim);
```
Тут библиотека должна упоминаться по полному пути а не по имени, для этого надо использовать `link(_whole)_rlib`, хотя и имя неподходящее, см https://github.com/rust-lang/rust/pull/99773.